homework02/rudolf/string.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include <iostream>
+#include <memory>
+#include <algorithm>
+
+std::size_t strlen(const char* chars);
+
+class String
+{
+public:
+	String();
+
+	String(const char* chars,std::size_t capacity);
+
+	String(const char* chars):String(chars,strlen(chars)){};
+	
+	String(const String& str):String(str.c_str()){};
+	
+	String operator=(String str);
+	
+	const bool empty()const {return mSize==0;}
+
+	char& operator[](int idx)const { return mString[idx];}
```

Comment:
This line cannot build, can it? You return a non-const reference to a member, while the method is const